internal/registry/test/mock_registry.go

```diff
@@ -167,9 +172,22 @@ func mockRegHandler() http.Handler {
 			location = fmt.Sprintf("file://%s/%s", wd, location)
 		}
 
-		w.Header().Set("X-Terraform-Get", location)
-		w.WriteHeader(http.StatusNoContent)
-		// no body
+		switch strings.HasPrefix(matches[0], "alpha/") {
```

Comment:
The `alpha` identifier here is repeated at least twice across the code and I assume it goes away at some point, which will break the tests. Maybe there should be a flag for it?

Reply:
You are right the `alpha` is an indicator of the alpha registry protocol to mock the response accordingly. I decided not to make over-invasive changes/optimisations to prevent blocking registry delivery. Please feel free to raise your ideas about refactoring - we will definitely come back to them after the stable release. Thanks!

Reply:
Thanks for the suggestion - I implemented flags as variadic arguments to the mock registry initialiser :)